torchtext/datasets/iwslt2016.py

```diff
@@ -0,0 +1,331 @@
+        },
+        'tst2013': {
+            ('en', 'ar'): 1169,
+            ('en', 'de'): 993,
+            ('en', 'fr'): 1026,
+            ('en', 'cs'): 1327,
+            ('ar', 'en'): 1169,
+            ('fr', 'en'): 1026,
+            ('de', 'en'): 993,
+            ('cs', 'en'): 1327
+        },
+        'tst2014': {
+            ('en', 'ar'): 1107,
+            ('en', 'de'): 1305,
+            ('en', 'fr'): 1305,
+            ('ar', 'en'): 1107,
+            ('fr', 'en'): 1305,
+            ('de', 'en'): 1305
+        }
+    },
+    'test': {
+        'dev2010': {
+            ('en', 'ar'): 887,
+            ('en', 'de'): 887,
+            ('en', 'fr'): 887,
+            ('en', 'cs'): 480,
+            ('ar', 'en'): 887,
+            ('fr', 'en'): 887,
+            ('de', 'en'): 887,
+            ('cs', 'en'): 480
+        },
+        'tst2010': {
+            ('en', 'ar'): 1569,
+            ('en', 'de'): 1565,
+            ('en', 'fr'): 1664,
+            ('en', 'cs'): 1511,
+            ('ar', 'en'): 1569,
+            ('fr', 'en'): 1664,
+            ('de', 'en'): 1565,
+            ('cs', 'en'): 1511
+        },
+        'tst2011': {
+            ('en', 'ar'): 1199,
+            ('en', 'de'): 1433,
+            ('en', 'fr'): 818,
+            ('en', 'cs'): 1013,
+            ('ar', 'en'): 1199,
+            ('fr', 'en'): 818,
+            ('de', 'en'): 1433,
+            ('cs', 'en'): 1013
+        },
+        'tst2012': {
+            ('en', 'ar'): 1702,
+            ('en', 'de'): 1700,
+            ('en', 'fr'): 1124,
+            ('en', 'cs'): 1385,
+            ('ar', 'en'): 1702,
+            ('fr', 'en'): 1124,
+            ('de', 'en'): 1700,
+            ('cs', 'en'): 1385
+        },
+        'tst2013': {
+            ('en', 'ar'): 1169,
+            ('en', 'de'): 993,
+            ('en', 'fr'): 1026,
+            ('en', 'cs'): 1327,
+            ('ar', 'en'): 1169,
+            ('fr', 'en'): 1026,
+            ('de', 'en'): 993,
+            ('cs', 'en'): 1327
+        },
+        'tst2014': {
+            ('en', 'ar'): 1107,
+            ('en', 'de'): 1305,
+            ('en', 'fr'): 1305,
+            ('ar', 'en'): 1107,
+            ('fr', 'en'): 1305,
+            ('de', 'en'): 1305
+        }
+    }
+}
+
+SET_NOT_EXISTS = {
+    ('en', 'ar'): [],
+    ('en', 'de'): [],
+    ('en', 'fr'): [],
+    ('en', 'cs'): ['tst2014'],
+    ('ar', 'en'): [],
+    ('fr', 'en'): [],
+    ('de', 'en'): [],
+    ('cs', 'en'): ['tst2014']
+}
+
+
+def _read_text_iterator(path):
+    with io.open(path, encoding="utf8") as f:
+        for row in f:
+            yield row
+
+
+def _clean_xml_file(f_xml):
+    f_txt = os.path.splitext(f_xml)[0]
+    with codecs.open(f_txt, mode='w', encoding='utf-8') as fd_txt:
+        root = ET.parse(f_xml).getroot()[0]
+        for doc in root.findall('doc'):
+            for e in doc.findall('seg'):
+                fd_txt.write(e.text.strip() + '\n')
+
+
+def _clean_tags_file(f_orig):
```

Comment:
As a follow-up this can lead to a generic dataset_utils.py function that can be used in multiple contexts.